logstash-core/lib/logstash/agent.rb

```diff
@@ -131,8 +133,42 @@ def shutdown
     shutdown_pipelines
   end
 
-  def node_uuid
-    @node_uuid ||= SecureRandom.uuid
+  def id
+    return @id if @id
+
+    uuid = nil
+    if File.exists?(id_path)
```

Comment:
`File` doesn't work all the time. because of the File input, would you mind adding `::File` to make sure we use the root namespace.  :(